### ISSUE TITLE
Onboarding tour to support new users through first-run

### ISSUE THREAD

**Issue by sandysage** [mozilla-lockbox/lockbox-extension#534]

**Comment by changecourse** [comment]
@sandysage we'll need to break out this design into tasks together next week

**Comment by sandysage** [comment]
Onboarding tour looks good. It probably needs review of content and engineering feedback next.

**Comment by linuxwolf** [comment]
@changecourse Some comments and questions:

* Is it safe to assume the "trigger" for onboading is the same as it is today? (e.g., on install, or when clicking the toolbar icon when not onboarded)
* If the user, for whatever reason, quits midway through (closes the web tab, closes the whole window, quits Firefox), what is the expected re-engagement?
* on "Welcome", what is expected from "Have an account? Sign in"?  Is it effectively (after FxA signin) a bypass the rest of the rest of onboarding?
* on "Import":
  - There is a "Deselect all".  What's the opinion on a corresponding "Select all" in some fashion?
  - As discussed IRL, if the user opts to "skip this step for now" they can re-engage from the account dropdown.
  - Once the user imports, is it expected that option is never presented again (including across devices)?
* On "Mobile Access" -- employing phone numbers has potential security, privacy, and legal issues; I've heard this might not be there for awhile, but we need to tread very carefully here.

**Comment by changecourse** [comment]
> Is it safe to assume the "trigger" for onboading is the same as it is today? (e.g., on install, or when clicking the toolbar icon when not onboarded)

Yes. This is also why there is a secondary hyperlink for sign-in if someone gets to this screen on a fresh install or separate Fx desktop experience but already has a Lockbox set up.

> If the user, for whatever reason, quits midway through (closes the web tab, closes the whole window, quits Firefox), what is the expected re-engagement?

One of three possibilities worth discussing based on value vs. LOE...
1. Preserve their position and bring them back to the corresponding section they left of on
2. Restart them back to the beginning, and don't save any work that has been done with previous import (so we don't have to worry about duplicates on import)
3. The only required step of onboarding is FxA... so maybe we use that as the flag... Authenticated? Bypass all other onboarding steps and end up on the home screen for new session... Not authenticated? Restart onboarding from step one. 

> On "Welcome", what is expected from "Have an account? Sign in"? Is it effectively (after FxA signin) a bypass the rest of the rest of onboarding?

Correct.

> On "Import:"  There is a "Deselect all". What's the opinion on a corresponding "Select all" in some fashion?

Agreed, but only to replace the "Deselect all" text in situations where it has been clicked... effectively turning that button into a toggle? All logins are selected by default, so I think this makes sense... 

> As discussed IRL, if the user opts to "skip this step for now" they can re-engage from the account dropdown.

Correct.

> Once the user imports, is it expected that option is never presented again (including across devices)?

Good question... my thought would be that we'd want to provide that option at all times (though buried within the dropdown, but I foresee some use cases involving a user importing some logins as part of a trial period, then being willing to bring in others at a later date... I do realize that this creates a heightened risk of duplicates... which we'd need to decide how to address gracefully. 

> On "Mobile Access" -- employing phone numbers has potential security, privacy, and legal issues; I've heard this might not be there for awhile, but we need to tread very carefully here.

Agreed... mostly envisioned as a future consideration for value, but needs more exploration and discussion with legal before pursuing.
